src/main/java/com/cegeka/academy/service/userChallenge/UserChallengeServiceImpl.java

```diff
@@ -26,4 +34,34 @@
         return userChallengeList.stream().map(userChallenge -> UserChallengeMapper.convertUserChallengeToUserChallengeDTO(userChallenge)).collect(Collectors.toList());
 
     }
+
+    @Override
+    public UserChallenge rateUser(UserChallengeDTO userChallengeDTO, Long ownerId, Long userChallengeId) throws WrongOwnerException {
+
+        UserChallenge userChallenge = new UserChallenge();
+        Optional<UserChallenge> userChallengeOptional = userChallengeRepository.findById(userChallengeDTO.getId());
+
+        if(userChallengeOptional.isPresent()) {
+           userChallenge = userChallengeOptional.get();
+        } else {
+            userChallengeOptional.orElseThrow(NoSuchElementException::new);
+        }
+        userChallenge.setPoints(userChallengeDTO.getPoints());
+
+        Challenge challenge = new Challenge();
+        Optional<Challenge> challengeOptional = challengeRepository.findById(userChallenge.getChallenge().getId());
+
```

Comment:
Aceasta verificare nu mai are sens. Daca te duci la baza de date si iei UserChallenge care contine un Challenge not null, este clar ca exista in baza de date